### PR TITLE
Properly raise parser error on size mismatch of code expressions

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: lib/fizzy/parser.cpp [hunk 1]
```diff
@@ -258,9 +262,9 @@ inline parser_result<Code> parse(const uint8_t* pos, const uint8_t* end)
         if (local_count > std::numeric_limits<uint32_t>::max())
             throw parser_error{"too many local variables"};
     }
-    std::get<0>(result).local_count = static_cast<uint32_t>(local_count);
+    result.local_count = static_cast<uint32_t>(local_count);
 
-    return result;
+    return {result, pos3};
```
Comment (axic) [inline lib/fizzy/parser.cpp]: Should use `std::move(result)` here?

Comment (gumb0) [reply lib/fizzy/parser.cpp]: Yes, I think it will be copied otherwise

File: lib/fizzy/parser.cpp [hunk 2]
```diff
@@ -249,7 +249,11 @@ inline parser_result<Code> parse(const uint8_t* pos, const uint8_t* end)
 
     const auto [locals_vec, pos2] = parse_vec<Locals>(pos1, end);
 
-    auto result = parse_expr(pos2, end);
+    auto [result, pos3] = parse_expr(pos2, end);
```
Comment (gumb0) [inline lib/fizzy/parser.cpp]: Could name it `[code, pos3]` not confuse with the whole tuple's type `parser_result`

Comment (axic) [reply lib/fizzy/parser.cpp]: Yeah, I've used that initially, but realised most other functions use `result`. I can change it.

Comment (axic) [reply lib/fizzy/parser.cpp]: Changed.

File: lib/fizzy/parser.cpp (hunk 1 above)
Comment (axic) [reply lib/fizzy/parser.cpp]: Done.

**Pull request merged**